awx/ui_next/src/screens/Job/JobOutput/shared/OutputToolbar.jsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import styled from 'styled-components';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { shape, func } from 'prop-types';
+import {
+  DownloadIcon,
+  RocketIcon,
+  TrashAltIcon,
+} from '@patternfly/react-icons';
+import { Badge as PFBadge, Button, Tooltip } from '@patternfly/react-core';
+import VerticalSeparator from '@components/VerticalSeparator';
+import DeleteButton from '@components/DeleteButton';
+import LaunchButton from '@components/LaunchButton';
+
+const BadgeGroup = styled.div`
+  margin-left: 20px;
+  height: 18px;
+  display: inline-flex;
+`;
+
+const Badge = styled(PFBadge)`
+  align-items: center;
+  display: flex;
+  justify-content: center;
+  margin-left: 10px;
+  ${props =>
+    props.color
+      ? `
+  background-color: ${props.color}
+  color: white;
+  `
+      : null}
+`;
+
+const Wrapper = styled.div`
+  align-items: center;
+  display: flex;
+  flex-flow: row wrap;
+  font-size: 14px;
+`;
+
+function toHHMMSS(s) {
+  function pad(n) {
+    return `00${n}`.slice(-2);
+  }
+
+  const secs = s % 60;
+  s = (s - secs) / 60;
+  const mins = s % 60;
+  const hrs = (s - mins) / 60;
+
```

Comment:
You can probably get the JS `Date` object to do most of the work for you, e.g:

![Screenshot from 2020-02-10 10-05-58](https://user-images.githubusercontent.com/9753817/74161578-2519c680-4bed-11ea-830c-36d9166bb94a.png)

Reply:
Ooo, using the date object is much more readable!